syntax = "proto3";

service InventoryService {
    rpc getAll(GetAllProductsRequest) returns (stream Product) {}
    rpc add(AddProductRequest)        returns (Product) {}
    rpc get(GetProductRequest)        returns (Product) {}
    rpc delete(DeleteProductRequest)  returns (Response) {}
}

message Product {
    string product_id = 1;
    string name       = 2;
    float price       = 3;
    int32 stock       = 4;
}

message AddProductRequest {
    string name = 1;
    float price = 2;
    int32 stock = 3;
}

message GetAllProductsRequest {
}

message Response {
    bool is_successful  = 1;
    string message      = 2;
}

message GetProductRequest {
    string product_id = 1;
}

message DeleteProductRequest {
    string product_id = 1;
}